Mathlib/MeasureTheory/Measure/Haar/DistribChar.lean

```diff
@@ -0,0 +1,91 @@
+/-
+Copyright (c) 2024 Andrew Yang, Yaël Dillies, Javier López-Contreras. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang, Yaël Dillies, Javier López-Contreras
+-/
+import Mathlib.MeasureTheory.Measure.Haar.Unique
+
+/-!
+# The distributive character of Haar measures
+
+Given a group `G` acting by additive morphisms on a locally compact additive commutative group `A`,
+and an element `g : G`, one can pull back the Haar measure `μ` of `A`
+along the map `(g • ·) : A → A` to get another Haar measure `μ'` on `A`.
+By unicity of Haar measures, there exists some nonnegative real number `r` such that `μ' = r • μ`.
+We can thus define a map `distribHaarChar : G → ℝ≥0` sending `g` to its associated real number `r`.
+Furthermore, this number doesn't depend on the Haar measure `μ` we started with,
+and `distribHaarChar` is a group homomorphism.
+
+## See also
+
+[Zulip](https://leanprover.zulipchat.com/#narrow/channel/217875-Is-there-code-for-X.3F/topic/canonical.20norm.20coming.20from.20Haar.20measure/near/480050592)
+-/
+
+open MeasureTheory.Measure
+open scoped NNReal Pointwise ENNReal
+
+namespace MeasureTheory
+variable {G A : Type*} [Group G] [AddCommGroup A] [DistribMulAction G A] [TopologicalSpace A]
+  [IsTopologicalAddGroup A] [LocallyCompactSpace A] [ContinuousConstSMul G A] {g : G}
+
+variable (A) in
+/-- The distributive Haar character of a group `G` acting distributively on a group `A` is the
+unique positive real number `Δ(g)` such that `μ (g • s) = Δ(g) * μ s` for all Haar
+measures `μ : Measure A`, set `s : Set A` and `g : G`. -/
+@[simps -isSimp]
+noncomputable def distribHaarChar : G →* ℝ≥0 :=
+  letI := borel A
+  have : BorelSpace A := ⟨rfl⟩
+  {
+    toFun g := addHaarScalarFactor (DomMulAct.mk g • addHaar) (addHaar (G := A))
```

Comment:
```suggestion
  { toFun g := addHaarScalarFactor (DomMulAct.mk g • addHaar) (addHaar (G := A))
```
Same thing at the end of the proof.

Reply:
This is on purpose. This style ensures that fields can easily be interchanged without churning the `{}` delimiters. It has been handy to me many times already when doing refactors that affect structure fields

Reply:
I'll leave it as is. Let's see how it fares